lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -380,9 +384,37 @@ def max_compact_name_size
       50
     end
 
+    def inspect_with_allocation_control(slice, memory_limit)
+      x = Thread.current
+      
+      start_alloc_size = ObjectSpace.memsize_of_all
+      
+      trace = TracePoint.new(:c_call, :call) do |tp|
+        curr_alloc_size = ObjectSpace.memsize_of_all
+        
+        if(curr_alloc_size - start_alloc_size > 1e6*memory_limit)
+          
+          trace.disable
+          x.raise MemoryLimitError, "Out of memory: evaluation took longer than 10mb." if x.alive?
+        end
+      end
+
+      trace.enable
+      slice.inspect
+      trace.disable 
+    rescue MemoryLimitError
+      return nil
```

Comment:
Do you think it makes sense to add some logging for such cases? Or return something very special